plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeSchemaSupport.java

```diff
@@ -185,6 +187,12 @@ public static boolean isDeletionVectorEnabled(MetadataEntry metadataEntry, Proto
     public static ColumnMappingMode getColumnMappingMode(MetadataEntry metadata, ProtocolEntry protocolEntry)
     {
         if (protocolEntry.supportsReaderFeatures() || protocolEntry.supportsWriterFeatures()) {
+            if (protocolEntry.writerFeaturesContains(ICEBERG_COMPATIBILITY_V1_FEATURE_NAME) || protocolEntry.writerFeaturesContains(ICEBERG_COMPATIBILITY_V2_FEATURE_NAME)) {
+                String columnMappingMode = metadata.getConfiguration().get(COLUMN_MAPPING_MODE_CONFIGURATION_KEY);
```

Comment:
why not use the "safe" way - similar to what is done a few lines below in the codebase ?

```
String columnMappingMode = metadata.getConfiguration().getOrDefault(COLUMN_MAPPING_MODE_CONFIGURATION_KEY, "none");
        return Enums.getIfPresent(ColumnMappingMode.class, columnMappingMode.toUpperCase(ENGLISH)).or(ColumnMappingMode.UNKNOWN);
```

Reply:
I avoided reusing the logic because the expected logic differs. Iceberg UniForm logic expects that `name` column mapping **_always_** exists as **_lowercase_**. 

I think we should fix the existing case-insensitive matching (toUpperCase). The protocol doesn't mention uppercase column mapping mode.